src/semgrep_agent/meta.py

```diff
@@ -113,6 +113,100 @@ def to_dict(self) -> Dict[str, Any]:
         }
 
 
+@dataclass
+class EnvVarMeta(GitMeta):
+    """Gather metadata only from local environment variables."""
+
+    config: str
+    environment: str = field(default="env-vars", init=False)
+
+    @cachedproperty
+    def event_name(self) -> Optional[str]:
+        return os.getenv("SEMGREP_EVENT_NAME")
+
+    @cachedproperty
+    def repo_name(self) -> str:
+        return os.getenv("SEMGREP_REPO_NAME") or "unset-SEMGREP_REPO_NAME"
+
+    @cachedproperty
+    def repo_url(self) -> Optional[str]:
+        return os.getenv("SEMGREP_REPO_URL")
+
+    @cachedproperty
+    def commit_sha(self) -> Optional[str]:
+        return os.getenv("SEMGREP_COMMIT_SHA")
+
+    @cachedproperty
+    def head_ref(self) -> Optional[str]:
+        return os.getenv("SEMGREP_COMMIT_SHA")
+
+    @cachedproperty
+    def branch(self) -> Optional[str]:
+        return os.getenv("SEMGREP_BRANCH")
+
+    @cachedproperty
+    def ci_job_url(self) -> Optional[str]:
+        return os.getenv("SEMGREP_JOB_URL")
+
+    @cachedproperty
+    def pr_id(self) -> Optional[str]:
+        return os.getenv("SEMGREP_PR_ID")
+
+    @cachedproperty
+    def pr_author_image_url(self) -> Optional[str]:
+        return os.getenv("SEMGREP_PR_AUTHOR_IMAGE_URL")
+
+    @cachedproperty
+    def pr_author_username(self) -> Optional[str]:
+        return os.getenv("SEMGREP_PR_AUTHOR_USERNAME")
+
+    @cachedproperty
+    def commit_author_image_url(self) -> Optional[str]:
+        return os.getenv("SEMGREP_COMMIT_AUTHOR_IMAGE_URL")
+
+    @cachedproperty
+    def commit_author_username(self) -> Optional[str]:
+        return os.getenv("SEMGREP_COMMIT_AUTHOR_USERNAME")
+
+    @cachedproperty
+    def commit_author_email(self) -> Optional[str]:
+        return os.getenv("SEMGREP_COMMIT_AUTHOR_EMAIL")
+
+    @cachedproperty
+    def commit_author_name(self) -> Optional[str]:
+        return os.getenv("SEMGREP_COMMIT_AUTHOR_NAME")
+
+    @cachedproperty
+    def commit_title(self) -> Optional[str]:
+        return os.getenv("SEMGREP_COMMIT_TITLE")
+
+    @cachedproperty
+    def pr_title(self) -> Optional[str]:
+        return os.getenv("SEMGREP_PR_TITLE")
+
+    def to_dict(self) -> Dict[str, Any]:
```

Comment:
Is there a way to get around this being a duplicate of the GitMeta to_dict? I could see us updating one and forgetting to update the other/ seems like a good place for code deduplication

Reply:
Yeah I should fold these into the GitMeta class on reflection. Thanks for the feedback